server/client.go

```diff
@@ -2523,8 +2574,18 @@ func (c *client) deliverMsg(sub *subscription, subject, mh, msg []byte, gwrply b
 	}
 
 	// Queue to outbound buffer
-	client.queueOutbound(mh)
-	client.queueOutbound(msg)
+	if client.flags.isSet(wsClient) {
```

Comment:
I wonder if we should make this its own handling instead of so many branches here.

Reply:
It is only 1 "if", or are you suggesting all the places where I deal with websocket clients? (readLoop, enqueue and deliverMsg, all these 3 duplicated)? Note that I may actually remove the changes here and in enqueueProto if I implement the framing at the level of flushOutbound instead of here.

Reply:
What do you think?

Reply:
So I would vote for keeping the "if" statements.